particles/Tutorial/Javascript/Demo/TTTGame.arcs

```diff
@@ -0,0 +1,138 @@
+// Tutorial Demo
+// Shows how to use Arcs features to create a tic-tac-toe game.
+import 'TTTGameSchemas.arcs'
+import 'TTTGameData.arcs'
+
+particle Game in './source/TTTGame.js'
+  inout Person {name, avatar, id} playerOne
+  inout Person {name, avatar, id} playerTwo
+  inout GameState gameState
+  inout Move {move} playerOneMove
+  inout Move {move} playerTwoMove
+  inout Event {type, move} event
+  check playerOne is private
+  check playerTwo is private
+  consume root
+    provide gameSlot
+
+particle TicTacToeBoard in './source/TTTBoard.js'
+  in GameState {board, currentPlayer} gameState
+  out Event {type, move} event
```

Comment:
My rationale for leaving event as a singleton instead of a collection is that collections don't have an order. So to have a stream of events coming in means a timestamp will have to be added and sorting will have to be done, increasing the mental overhead. This is probably biased because I don't like collections; interacting with them just feels a bit clunky.

Reply:
I implemented it just to see how bad it would be, and it's not bad at all. I was just scared. Javascript magic makes it fairly trivial.